ExampleApp-HandlingErrors/ViewController.swift

```diff
@@ -29,9 +29,60 @@ class ViewController: UIViewController {
         diceImage.layer.shadowRadius = 2
         diceImage.layer.shadowOffset = .zero
     }
+    
+    private var isRolling = false {
+        didSet {
+            UIView.animate(withDuration: 0.33) { [unowned self] in
+                diceImage.alpha = isRolling ? 0.5 : 1.0
+                diceImage.transform = isRolling ? CGAffineTransform(scaleX: 0.5, y: 0.5) :
+                    CGAffineTransform.identity
+            }
+            rollDiceButton.isEnabled = !isRolling
```

Comment:
This is the type of logic that ideally _should_ be in the viewModel, as it is way easier to test there.

Reply:
Agreed. I assumed this is the reason you made the VM the center of all state management. I wanted to confirm. Thanks!